src/cli-validator/utils/processConfiguration.js

```diff
@@ -168,11 +168,19 @@ const validateConfigObject = function(configObject, chalk) {
   return configObject;
 };
 
-const getConfigObject = async function(defaultMode, chalk) {
+const getConfigObject = async function(defaultMode, chalk, configFileOverride) {
   let configObject;
-  // search up the file system for the first instance
-  // of the config file
-  const configFile = await findUp('.validaterc');
+
+  let configFile;
+
+  if (configFileOverride) {
+    // user explicitly provided a config file, use it instead of .validaterc
+    configFile = configFileOverride;
+  } else {
+    // search up the file system for the first instance
+    // of the config file
+    configFile = await findUp('.validaterc');
+  }
```

Comment:
Might I suggest a slightly different implementation here? My concern is the error handling if the file at the specific path doesn't exist.

Although it's not exactly the intended function, `find-up` can be used to determine existence of a file if given a full path (and it won't look anywhere else). I propose something like this:

```js
  let configObject;

  // if user explicitly provided a config file, use it instead of .validaterc
  const configFileName = configFileOverride || '.validaterc';

  // search up the file system for the first instance
  // of '.validaterc' or,
  // if a config file override is passed in, use find-up
  // to verify existence of the file
  const configFile = await findUp(configFileName);

  // if the user does not have a config file, run in default mode and warn them
  // (findUp returns null if it does not find a file)
  if (configFile === null && !defaultMode) {
    if (configFileOverride) {
      console.log(
        '\n' +
          chalk.yellow('[Warning]') +
          ` No file found at specified path: ${chalk.underline(
            `${configFileOverride}`
          )}. The validator will run in ` +
          chalk.bold.cyan('default mode.')
      );
    } else {
      console.log(
        '\n' +
          chalk.yellow('[Warning]') +
          ` No ${chalk.underline(
            '.validaterc'
          )} file found. The validator will run in ` +
          chalk.bold.cyan('default mode.')
      );
      console.log(`To configure the validator, create a .validaterc file.`);
    }
    defaultMode = true;
  }
```

That keeps the code very similar, adds an extra layer of protection against a missing file, and (arguably most importantly) makes the error messages specific to the problem. Perhaps there's a better way to write it or better comments so feel free to make changes but that's the pattern I would prefer. Let me know what you think!

Reply:
I like your implementation, will change to that.